arch/riscv/core/smp.c

```diff
@@ -164,6 +165,20 @@ void arch_spin_relax(void)
 }
 #endif
 
+#if DT_HAS_COMPAT_STATUS_OKAY(sifive_clint0)
```

Comment:
Alternatively, we can introduce a Kconfig for this instead, i.e. `CONFIG_RISCV_SOC_IPI_CLINT`